From 26c3dc3986760483625262a4a3564f1696f829b3 Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Mon, 9 Mar 2015 12:41:43 +0100
Subject: [PATCH] metatree: always clear fd to -1

meta_tree_clear clears fd to 0 currently, however it should be set to -1
to avoid risk that fd 0 is closed, when meta_tree_clear is called twice.

https://bugzilla.gnome.org/show_bug.cgi?id=598561
---
 metadata/metatree.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/metadata/metatree.c b/metadata/metatree.c
index 0fe29a1..4c6d5f0 100644
--- a/metadata/metatree.c
+++ b/metadata/metatree.c
@@ -258,7 +258,7 @@ meta_tree_clear (MetaTree *tree)
   if (tree->fd != -1)
     {
       close (tree->fd);
-      tree->fd = 0;
+      tree->fd = -1;
     }
 }
 
-- 
2.5.0

